### ISSUE TITLE
Cursor extremely difficult to see on dark theme 

### ISSUE THREAD

**Issue by cddesja** [rstudio/rstudio#2685]
### System details

    RStudio Edition : Desktop
    RStudio Version :  1.1.447
    OS Version      :  macOS 10.13.4 
    R Version       :  3.4.4

### Steps to reproduce the problem

Select a dark theme, such as Dracula. 

### Describe the problem in detail

The cursor when hovering over the code is black and extremely difficult to locate. At least for, dark themes, are unusable. This is likely a function of the high res retina display I have. 

### Describe the behavior you expected

VS Code, e.g., when handling a dark theme, turns the cursor to white. Is it possible to do this with RStudio? 

See the mouse over the D in on line 145 in the red rectangle in RStudio:
<img width="1680" alt="rstudio" src="https://user-images.githubusercontent.com/1608939/39141709-27649e16-46ee-11e8-8b93-767c048c29b6.png">

See the mouse over the D in on line 145 in the red rectangle in VS code:
<img width="1680" alt="vscode" src="https://user-images.githubusercontent.com/1608939/39141724-34703c3c-46ee-11e8-8edb-0ae743a5a290.png">

Please ignore the size of the cursors as they are actually the same in both apps but for whatever reason appears larger when taking a screenshot for VS Code.

**Comment by ronblum** [comment]
@cddesja That is a good point. We'll review it during ongoing development of RStudio.

**Comment by cddesja** [comment]
I can verify that this doesn't happen with the the dark themes on Windows. The cursor is white. So, this  is probably just a Mac (and possibly Linux) issue.

**Comment by javierluraschi** [comment]
Looks like this has been the default cursor behavior since 1.0:

<img width="656" alt="screen shot 2018-06-11 at 3 08 26 pm" src="https://user-images.githubusercontent.com/3478847/41236560-56d967aa-6d89-11e8-8983-328c9d7a0e83.png">

**Comment by samhinshaw** [comment]
I mentioned this [over on support.rstudio.com](https://support.rstudio.com/hc/en-us/community/posts/213681648-Cursor-in-dark-theme-hard-to-see-Macos-) last year, but this bug cropped up in two Electron-based macOS apps, Hyper and VS Code. 

- [Fix for Hyper](https://github.com/zeit/hyper/issues/1045)
- [Fix for VS Code](https://github.com/Microsoft/vscode/issues/754)

I was unable to implement the workaround previously, but it seems to be working with the new rendering engine in RStudio 1.2. The workaround is to specify your own [text cursor](https://developer.mozilla.org/en-US/docs/Web/CSS/cursor) with the following css:

```css
* {
  cursor: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABAAAAAQCAQAAAC1+jfqAAAAL0lEQVQoz2NgCD3x//9/BhBYBWdhgFVAiVW4JBFKGIa4AqD0//9D3pt4I4tAdAMAHTQ/j5Zom30AAAAASUVORK5CYII=')
      5 8,
    text;
}
```

**Note**: to edit a theme's CSS in your compiled RStudio distribution, open Applications and right-click on RStudio -> [Show Package Contents]. Then Navigate to `Contents/Resources/www/rstudio` and edit the `<string>.cache.css` file corresponding to the theme you use... this might involve some guesswork.

Perhaps this could be injected into compiled themes' CSS for macOS distributions of RStudio? I am not sure how it would affect other platforms. Additionally, the original fix I found had multiple DPI options which did not seem to work properly for RStudio.

**Comment by cddesja** [comment]
@samhinshaw, thanks for the work around. 

Outside of literally guessing which `<string>.cache.css` goes with which theme, does anyone have an idea of how to identify a theme's corresponding CSS? For example, if I wanted to do this with the Dracula theme, which CSS would I change?

**Comment by samhinshaw** [comment]
@cddesja You need to be able to use the rendering engine's dev tools to inspect the layout. This will tell you which files the styles are coming from. However, the ability to do so is currently disabled: #2133 

You could probably check with RStudio 1.1 or wait until Qt 5.11 is merged and try a daily build.

**Comment by gtritchie** [comment]
@MariaSemple this is firmly in your territory. Would be a nice improvement to consider on macOS.

**Comment by TrashBirdEcology** [comment]
I would love to see this resolved, if it hasn't already been done.

**Comment by GreenGrassBlueOcean** [comment]
Hi I am currently using Rstudio 1.4.904 and this is still a huge issue. Especially when the cursor is in selection mode e,g, like: I. 
I am using Rstudio theme: Mordern, Editor Theme: idle fingers

**Comment by svalvaro** [comment]
Any updates on this? Still an annoying behavior after 5 years. It's also applicable to Rstudio Workbench.

**Comment by GreenGrassBlueOcean** [comment]
yes would be wonderful if this would be fixed!, so annoying to look for the cursor in the help and build panes of rstudio
